core/src/test/java/com/sequenceiq/cloudbreak/service/stack/flow/UpdateNodeCountValidatorTest.java

```diff
@@ -137,25 +139,39 @@ private static Stream<Arguments> testValidateScalabilityOfInstanceGroupData() {
     public void testValidateStatusForStartHostGroup(
             Status status,
             Optional<String> errorMessageSegment) {
+        StackDto stackdto = mock(StackDto.class);
+        StackView stackView = mock(StackView.class);
+        setupMocksForStopStartInstanceGroupValidation(stackdto);
+        CmTemplateProcessor cmTemplateProcessor = mock(CmTemplateProcessor.class);
+        when(cmTemplateProcessorFactory.get(anyString())).thenReturn(cmTemplateProcessor);
+        List<String> stackUnavaiable = new ArrayList<>();
+        stackUnavaiable.add("No check required");
+        when(stackdto.getStack()).thenReturn(stackView);
         Stack stack = mock(Stack.class);
+        when(stack.getStack()).thenReturn(stackView);
         Cluster cluster = mock(Cluster.class);
         when(stack.getName()).thenReturn("master-stack");
+        when(stackdto.getStack().getName()).thenReturn("master-stack");
         when(stack.getStatus()).thenReturn(status);
+        when(stackdto.getStack().getStatus()).thenReturn(status);
         when(stack.getCluster()).thenReturn(cluster);
         when(cluster.getName()).thenReturn("master-stack");
         if (status == AVAILABLE) {
             when(stack.isAvailable()).thenReturn(true);
+            when(stackdto.getStack().isAvailable()).thenReturn(true);
```

Comment:
Please create a unit test for the new scenario that is being introduced: Unhealthy services on dependent hosts for above 7.2.12, and below 7.2.12 etc

Reply:
Yes, I will cover that